gator/fragments.py

```diff
@@ -1,64 +1,59 @@
 """Retrieve and count the contents of a file"""
 
 from pathlib import Path
-import re
+
+import commonmark
 
 from gator import util
 
 FILE_SEPARATOR = "/"
 
-# References:
-# https://stackoverflow.com/questions/18568105/how-match-a-paragraph-using-regex
-# https://stackoverflow.com/questions/13531204/how-to-match-paragraphs-in-text-with-regx
-
-CODE_FENCE_MARKER = "```"
-GATORGRADER_REPLACEMENT = "GATORGRADER_REPLACEMENT"
-PARAGRAH_RE = r"(.+?\n\n|.+?$)"
-SECTION_MARKER = "#"
-
 NEWLINE = "\n"
 NOTHING = ""
 SPACE = " "
 
-DOUBLE_NEWLINE = NEWLINE * 2
-
 
-def is_paragraph(candidate):
-    """Determines if a writing candidate is a paragraph"""
-    # remove whitespace surrounding candidate paragraph
-    candidate = candidate.strip()
-    # if the paragraph is a markdown header, it is not a paragraph
-    if candidate.startswith(SECTION_MARKER):
-        return False
-    # if the paragraph is a fenced code block, it is not a paragraph
-    if candidate.startswith(CODE_FENCE_MARKER):
-        return False
-    # there may be other edge cases that should be added here in the
-    # future -- what other structures look like paragraphs but should
-    # not be?
-    # if nothing has returned by now, the candidate must be a paragraph
-    return True
+def is_single_subnode(subnode):
+    """Determines if a node is single"""
+    return (
+        # Excludes the following subnode types
```

Comment:
This documentation comment:
* Does not belong here (documentation should be placed in the docstring above).
* Is not fully accurate (what are they excluded from?)